x-pack/plugins/ui_actions_enhanced/public/drilldowns/url_drilldown/url_template.ts

```diff
@@ -5,130 +5,147 @@
  * 2.0.
  */
 
-import { create as createHandlebars, HelperDelegate, HelperOptions } from 'handlebars';
+import type { HelperDelegate, HelperOptions } from 'handlebars';
+import type * as Handlebars from 'handlebars';
 import { encode, RisonValue } from 'rison-node';
 import dateMath from '@elastic/datemath';
 import moment, { Moment } from 'moment';
 import numeral from '@elastic/numeral';
 import { url } from '../../../../../../src/plugins/kibana_utils/public';
 
-const handlebars = createHandlebars();
-
-function createSerializationHelper(
-  fnName: string,
-  serializeFn: (value: unknown) => string
-): HelperDelegate {
-  return (...args) => {
-    const { hash } = args.slice(-1)[0] as HelperOptions;
-    const hasHash = Object.keys(hash).length > 0;
-    const hasValues = args.length > 1;
-    if (hasHash && hasValues) {
-      throw new Error(`[${fnName}]: both value list and hash are not supported`);
-    }
-    if (hasHash) {
-      if (Object.values(hash).some((v) => typeof v === 'undefined'))
-        throw new Error(`[${fnName}]: unknown variable`);
-      return serializeFn(hash);
-    } else {
-      const values = args.slice(0, -1) as unknown[];
-      if (values.some((value) => typeof value === 'undefined'))
-        throw new Error(`[${fnName}]: unknown variable`);
-      if (values.length === 0) throw new Error(`[${fnName}]: unknown variable`);
-      if (values.length === 1) return serializeFn(values[0]);
-      return serializeFn(values);
-    }
-  };
-}
+let handlebarsPromise: Promise<typeof Handlebars>;
+async function getHandlebars(): Promise<typeof Handlebars> {
```

Comment:
Actually, what do you think about moving all of this code into a lazy-loaded module and then importing that, instead of lazy importing `handlebars`, i.e.:

```suggestion
const getHandlebars = () => import('./handlebars').getHandlebars();
```

Reply:
I agree 👍 good point